app/src/main/java/com/bignerdranch/android/geoquiz/views/QuizActivity.kt

```diff
@@ -0,0 +1,181 @@
+            if (data == null) {
+                return
+            }
+            isCheater = CheatActivity.wasAnswerShown(data)
+        }
+    }
+
+    public override fun onSaveInstanceState(savedInstanceState: Bundle) {
+        super.onSaveInstanceState(savedInstanceState)
+        Timber.d("onSaveInstanceState")
+        savedInstanceState.putInt(KEY_INDEX, quizViewModel.currentIndex)
+    }
+
+    public override fun onStart() {
+        super.onStart()
+        Timber.d("onStart() called")
+    }
+
+    public override fun onPause() {
+        super.onPause()
+        Timber.d("onPause() called")
+    }
+
+    public override fun onResume() {
+        super.onResume()
+        Timber.d("onResume() called")
+    }
+
+    public override fun onStop() {
+        super.onStop()
+        Timber.d("onStop() called")
+    }
+
+    public override fun onDestroy() {
+        super.onDestroy()
+        Timber.d("onDestroy() called")
+    }
+
+    private fun updateQuestion() {
+        val actualQuestion = quizViewModel.currentQuestion
+        questionTextView!!.setText(quizViewModel.currentQuestionText)
+        setButtonsState(!resolvedQuestions.contains(actualQuestion))
+        showPercentage()
+    }
+
+    private fun updateView(isCorrect: Boolean) {
+        val actualQuestion = quizViewModel.currentQuestion
+        if (resolvedQuestions.contains(actualQuestion)) {
+            setButtonsState(false)
+        } else {
+            showToastCorrectly(isCorrect)
+            increaseCounters(isCorrect)
+            updateResolvedQuestions(actualQuestion)
+        }
+    }
+
+    private fun showToastCorrectly(isCorrect: Boolean) {
+        @StringRes
+        val answerTest = if (isCheater) R.string.judgment_toast else getAnswerText(isCorrect)
+        val toast = Toast.makeText(this, answerTest, Toast.LENGTH_SHORT)
+        toast.setGravity(Gravity.TOP or Gravity.CENTER_HORIZONTAL, 0, 0)
+        toast.show()
+    }
+
+    @StringRes
+    private fun getAnswerText(isCorrect: Boolean): Int {
+        return if (isCorrect) R.string.correct else R.string.incorrect
+    }
+
+    private fun increaseCounters(isCorrect: Boolean) {
+        if (isCorrect) {
+            countOfCorrectQuestionsResolved += 1
+        }
+        countOfQuestionsResolved += 1
+    }
+
+    private fun updateResolvedQuestions(actualQuestion: Question) {
+        resolvedQuestions.add(actualQuestion)
+        setButtonsState(false)
+    }
+
+    private fun setButtonsState(enabled: Boolean) {
+        questionButtonFalse!!.isEnabled = enabled
+        questionButtonTrue!!.isEnabled = enabled
+    }
+
+    private fun showPercentage() {
+        if (countOfQuestionsResolved == quizViewModel.questionBank.size) {
+            val percentage = "$countOfCorrectQuestionsResolved/$countOfQuestionsResolved"
+            Toast.makeText(this, "The average is: $percentage", Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    companion object {
```

Comment:
we usually put companion object at the beginning of the class, move it to the top